python/semantic_kernel/connectors/ai/ollama/services/ollama_chat_completion.py

```diff
@@ -0,0 +1,121 @@
+
+    Arguments:
+        ai_model_id {str} -- Ollama model name, see https://ollama.ai/library
+        url {Optional[Union[str, HttpUrl]]} -- URL of the Ollama server, defaults to http://localhost:11434/api/chat
+    """
+
+    url: HttpUrl = "http://localhost:11434/api/chat"
+    session: Optional[aiohttp.ClientSession] = None
+
+    async def complete_chat_async(
+        self,
+        messages: List[Dict[str, str]],
+        request_settings: OllamaChatRequestSettings,
+        **kwargs,
+    ) -> Union[str, List[str]]:
+        request_settings.messages = messages
+        request_settings.stream = False
+        if self.session:
+            async with self.session.post(str(self.url), json=request_settings.prepare_settings_dict()) as response:
+                response.raise_for_status()
+                response_object = await response.json()
+                return response_object.get("message", {"content": None}).get("content", None)
+        async with aiohttp.ClientSession() as session:
+            async with session.post(str(self.url), json=request_settings.prepare_settings_dict()) as response:
+                response.raise_for_status()
+                response_object = await response.json()
+                return response_object.get("message", {"content": None}).get("content", None)
+
+    async def complete_chat_stream_async(
+        self,
+        messages: List[Dict[str, str]],
+        settings: OllamaChatRequestSettings,
+        **kwargs,
+    ):
+        """
+        Streams a text completion using a Ollama model.
+        Note that this method does not support multiple responses.
+
+        Arguments:
+            prompt {str} -- Prompt to complete.
+            request_settings {OllamaChatRequestSettings} -- Request settings.
+
+        Yields:
+            str -- Completion result.
+        """
+        settings.messages = messages
+        settings.stream = True
+        if self.session:
+            async with self.session.post(str(self.url), json=settings.prepare_settings_dict()) as response:
+                response.raise_for_status()
+                async for line in response.content:
+                    body = json.loads(line)
+                    response_part = body.get("message", {"content": None}).get("content", None)
+                    yield response_part
+                    if body.get("done"):
+                        break
+        if not self.session:
+            async with aiohttp.ClientSession() as session:
+                async with session.post(str(self.url), json=settings.prepare_settings_dict()) as response:
+                    response.raise_for_status()
+                    async for line in response.content:
+                        body = json.loads(line)
+                        response_part = body.get("message", {"content": None}).get("content", None)
+                        yield response_part
+                        if body.get("done"):
+                            break
```

Comment:
Could this block be refactored using the `AsyncSession` mentioned before?

Reply:
I remember seeing this changes and now they are gone and the duplicated logic is back in.